torch_geometric/utils/augmentation.py

```diff
@@ -0,0 +1,190 @@
+    indicating the orders of original nodes after shuffling.
+
+    Args:
+        x (FloatTensor): The feature matrix.
+        training (bool, optional): If set to :obj:`False`, this operation is a
+            no-op. (default: :obj:`True`)
+
+    :rtype: (:class:`FloatTensor`, :class:`LongTensor`)
+
+    Example:
+
+        >>> x = torch.tensor([[0, 1, 2],
+        ...                   [3, 4, 5],
+        ...                   [6, 7, 8]], dtype=torch.float)
+        >>> x, node_perm = shuffle_node(x)
+        >>> x
+        tensor([[6., 7., 8.],
+                [3., 4., 5.],
+                [0., 1., 2.]])
+        >>> node_perm
+        tensor([2, 1, 0])
+    """
+    if not training:
+        perm = torch.arange(x.size(0))
+        return x, perm
+    perm = torch.randperm(x.size(0))
+    return x[perm], perm
+
+
+def mask_feature(x: Tensor, p: float = 0.5, column_wise: bool = True,
+                 training: bool = True) -> Tuple[Tensor, Tensor]:
+    r"""Randomly masks feature from the feature matrix
+    :obj:`x` with probability :obj:`p` using samples from
+    a Bernoulli distribution.
+
+    The method returns (1) the retained :obj:`x`, (2) the feature
+    mask, the same shape as :obj:`x`, indicating where features are retained.
+
+    Args:
+        x (FloatTensor): The feature matrix.
+        p (float, optional): Masking ratio. (default: :obj:`0.5`)
+        column_wise (bool, optional): If set to :obj:`True`, masks whole feature
+             columns with probability :obj:`p`.  (default: :obj:`True`)
+        training (bool, optional): If set to :obj:`False`, this operation is a
+            no-op. (default: :obj:`True`)
+
+    :rtype: (:class:`FloatTensor`, :class:`BoolTensor`)
+
+    Examples:
+
+        >>> # Masked features are column-wise sampled
+        >>> x = torch.tensor([[0, 1, 2],
+        ...                   [3, 4, 5],
+        ...                   [6, 7, 8]], dtype=torch.float)
+        >>> x, feat_mask = mask_feature(x)
+        >>> x
+        tensor([[0., 0., 2.],
+                [3., 0., 5.],
+                [6., 0., 8.]]),
+        >>> feat_mask
+        tensor([[ True, False,  True],
+                [ True, False,  True],
+                [ True, False,  True]])
+
+        >>> # Masked features are uniformly sampled
+        >>> x, feat_mask = mask_feature(x, column_wise=False)
+        >>> x
+        tensor([[0., 0., 0.],
+                [3., 0., 5.],
+                [0., 0., 8.]])
+        >>> feat_mask
+        tensor([[False, False, False],
+                [ True, False,  True],
+                [False, False,  True]])
+    """
+    if p < 0. or p > 1.:
+        raise ValueError(f'Masking ratio has to be between 0 and 1 '
+                         f'(got {p}')
+    if not training or p == 0.0:
+        return x, torch.ones_like(x, dtype=torch.bool)
+
+    if column_wise:
+        mask = torch.rand(x.size(1), device=x.device) >= p
+        mask = mask.tile(x.size(0), 1)
```

Comment:
```suggestion
        mask = mask.view(1, -1)
```

Reply:
I'm afraid this cannot broadcast `mask` to the shape the same as `x`

Reply:
Ok, just trying to understand why not?

Reply:
In this case
```python
x: shape (N, M)
mask: shape (M)

mask.view(1, -1): shape  (1, M)
mask.tile(x.size(0), 1): shape (N, M)
```

Reply:
Yes, so that should be broadcastable, right?

Reply:
but the returned mask would get the shape (1, M) that doesn't match the shape of `x` (N, M). We should return the mask with the same shape of `x`, make sense?

Reply:
They should be broadcastable according to this [pytorch](https://pytorch.org/docs/stable/notes/broadcasting.html#broadcasting-semantics) doc.
`masked_fill` works if `x` and `mask` are broadcastable.

Reply:
My bad, they are indeed broadcastable. But I still think we should use `mask.tile(x.size(0), 1)` rather than `mask.view(1, -1)` to make `x` and `x` the same shapes. WDYT?

Reply:
For large features with many nodes it might not be the most efficient. Lets maybe update the doc to mention that `mask` is broadcastable with x (and not necessarily the same shape). WDYT?

Reply:
Agreed. Didn't take the efficiency into consideration.